### PR TITLE
Barista/expose filter field outputs in quick filter

### PR DESCRIPTION
**Pull request by christian-fischer**
### <strong>Pull Request</strong>

<hr>
Hi, thank you for contributing to Barista with this pull request (PR).

To ensure a fast process and merging of your PR please make sure it fulfills the
coding standards and contribution guidelines.

- A feature proposal has been provided, discussed and approved first.
- There is a meaningful description of the issue in GitHub (Screenshots are
  often helpful).
- If the PR introduces breaking-changes or deprecations it matches the following
  guidelines.
  - The commit message follows our commit guidelines.
  - Tests for the changes have been added (for bug fixes / features).
  - Docs have been added / updated (for bug fixes / features).

Please choose the type appropriate for the changes below: <br>

#### Type of PR

Feature (non-breaking change which adds functionality)

#### Checklist

- [x] I have read the CONTRIBUTING doc and I follow the PR guidelines
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: libs/barista-components/experimental/quick-filter/src/quick-filter.html [hunk 1]
```diff
@@ -1,6 +1,8 @@
 <dt-filter-field
   [dataSource]="_filterFieldDataSource$ | async"
-  (filterChanges)="_filterFiledChanged($event)"
+  (inputChange)="inputChange.emit($event)"
+  (currentFilterChanges)="currentFilterChanges.emit($event)"
+  (filterChanges)="_filterFieldChanged($event)"
```
Comment (tomheller) [inline libs/barista-components/experimental/quick-filter/src/quick-filter.html:5]: @thomaspink I think there is a better way to proxy the event through, without emitting a new one. 
Can you please advise?

File: libs/barista-components/experimental/quick-filter/src/quick-filter.ts [hunk 2]
```diff
@@ -77,11 +86,19 @@ export class DtQuickFilterChangeEvent<T> extends DtFilterFieldChangeEvent<T> {}
...
+  /** Emits an event with the current value of the input field every time the user types. */
+  @Output() readonly inputChange = new EventEmitter<string>();
+
   /** Emits when a new filter has been added or removed. */
   @Output() readonly filterChanges = new EventEmitter<
     DtQuickFilterChangeEvent<T>
   >();
 
+  /** Emits when a part has been added to the currently active filter. */
+  @Output() readonly currentFilterChanges = new EventEmitter<
```
[the remark this answers is not in the document]
Comment (ffriedl89) [reply libs/barista-components/experimental/quick-filter/src/quick-filter.ts:98]: @christian-fischer Please reach out to @thomaspink for guidance or help :)

**Comment by christian-fischer** [comment]
update: 
- rebase to current upstream master
- incorporated changes suggested by @thomaspink 
- adapted tests to reflect those changes.

File: libs/barista-components/experimental/quick-filter/README.md [hunk 3]
```diff
@@ -46,6 +46,8 @@ sidebar you can leverage the `<dt-quick-filter-title>` and
 
 ## Outputs
 
-| Name            | Type                                     | Description                                                                                                 |
-| --------------- | ---------------------------------------- | ----------------------------------------------------------------------------------------------------------- |
-| `filterChanges` | `EventEmitter<DtQuickFilterChangeEvent>` | Event emitted when filters have been updated by user interaction. Wont be triggered by programmatic changes |
+| Name                   | Type                                                  | Description                                                                                                 |
+| ---------------------- | ----------------------------------------------------- | ----------------------------------------------------------------------------------------------------------- |
+| `filterChanges`        | `EventEmitter<DtQuickFilterChangeEvent>`              | Event emitted when filters have been updated by user interaction. Wont be triggered by programmatic changes |
```
Comment (tomheller) [inline libs/barista-components/experimental/quick-filter/README.md:51]: nitpick:
```suggestion
| `filterChanges`        | `EventEmitter<DtQuickFilterChangeEvent>`              | Event emitted when filters have been updated by user interaction. Wont be triggered by programmatic changes. |
```

**Comment by tomheller** [comment]
Attention to the cherry-picker: This PR contains a fix and a feat commit. 
Cherry pick the to the correct branches separately. I think we can do that as we need to cherry pick manually anyways (as this is not supported for forks yet).

**Comment by tomheller** [comment]
Thank you for contributing this fix. 
Cherry-picked to 7.x 0c44e64039c5d54b057151752553bc029f819228 and 7.0.x 90e08f028eac5d1398493446d9ccc9bb21da73d8

**Pull request merged**